roles/custom/matrix-bot-chatgpt/defaults/main.yml

```diff
@@ -0,0 +1,78 @@
+---
+# chatgpt is a bot for chatting to openAI chatgpt matrix bot
+# Project source code URL: https://github.com/jakecoppinger/matrix-chatgpt-bot
+
+matrix_bot_chatgpt_enabled: true
+
+matrix_bot_chatgpt_container_image_self_build: false
+matrix_bot_chatgpt_docker_repo: "https://github.com/matrixgpt/matrix-chatgpt-bot"
+matrix_bot_chatgpt_docker_repo_version: "{{ 'latest' if matrix_bot_chatgpt_version == 'latest' else matrix_bot_chatgpt_version }}"
+matrix_bot_chatgpt_docker_src_files_path: "{{ matrix_base_data_path }}/chatgpt/docker-src"
+
+matrix_bot_chatgpt_version: 1.1.1
+matrix_bot_chatgpt_docker_image: "{{ matrix_bot_chatgpt_docker_image_name_prefix }}matrixgpt/matrix-chatgpt-bot:{{ matrix_bot_chatgpt_version }}"
+matrix_bot_chatgpt_docker_image_name_prefix: "{{ 'localhost/' if matrix_bot_chatgpt_container_image_self_build else 'ghcr.io/' }}"
+matrix_bot_chatgpt_docker_image_force_pull: "{{ matrix_bot_chatgpt_docker_image.endswith(':latest') }}"
+
+matrix_bot_chatgpt_base_path: "{{ matrix_base_data_path }}/chatgpt"
+matrix_bot_chatgpt_config_path: "{{ matrix_bot_chatgpt_base_path }}/config"
+matrix_bot_chatgpt_data_path: "{{ matrix_bot_chatgpt_base_path }}/data"
+
+# A list of extra arguments to pass to the container
+matrix_bot_chatgpt_container_extra_arguments: []
+
+# List of systemd services that matrix-bot-chatgpt.service depends on
+matrix_bot_chatgpt_systemd_required_services_list: ['docker.service']
+
+# List of systemd services that matrix-bot-chatgpt.service wants
+matrix_bot_chatgpt_systemd_wanted_services_list: []
+
+# ChatGPT Settings
+matrix_bot_chatgpt_openai_email: '' # OPENAI_EMAIL=
+matrix_bot_chatgpt_openai_password: '' # OPENAI_PASSWORD=
+matrix_bot_chatgpt_openai_login_type: 'google' # OPENAI_LOGIN_TYPE="google"
+
+# Matrix Static Settings (required, see notes)
+# Defaults to "https://matrix.org"
+matrix_bot_chatgpt_matrix_homeserver_url: "{{ matrix_homeserver_container_url }}" # MATRIX_HOMESERVER_URL=
+# With the @ and :DOMAIN, ie @SOMETHING:DOMAIN, needs to be set, created manually beforehand.
+matrix_bot_chatgpt_matrix_bot_username: '@chatgpt:{{ matrix_domain }}' # MATRIX_BOT_USERNAME=
+# Set `MATRIX_BOT_PASSWORD` the bot will print an `MATRIX_ACCESS_TOKEN` to the terminal
+# or https://webapps.stackexchange.com/questions/131056/how-to-get-an-access-token-for-element-riot-matrix
+matrix_bot_chatgpt_matrix_access_token: '' # MATRIX_ACCESS_TOKEN=
+# Once `MATRIX_BOT_ACCESS_TOKEN` is set this is no longer used.
+matrix_bot_chatgpt_matrix_bot_password: '' # MATRIX_BOT_PASSWORD=
+
+# Matrix Configurable Settings Defaults (optional)
+matrix_bot_chatgpt_matrix_default_prefix: "!chatgpt " # MATRIX_DEFAULT_PREFIX= Leave prefix blank to reply to all messages, trailing space matters
+matrix_bot_chatgpt_matrix_default_prefix_reply: false # MATRIX_DEFAULT_PREFIX_REPLY=
+
+# Matrix Access Control (optional)
+matrix_bot_chatgpt_matrix_blacklist: ''
+matrix_bot_chatgpt_matrix_whitelist: ''
```

Comment:
Does this mean anyone (on any homeserver) can use the bot?

All our other bots default to "this homeserver only" and require manual configuration changes to make them work for external users. Perhaps we should do the same here.

Reply:
Yeah we could set this to :

```
matrix_bot_chatgpt_matrix_whitelist: ':{{ matrix_domain }}'
```

Reply:
Do we need `matrix_bot_chatgpt_matrix_blacklist: '*'` or something of that sort as well or is just the whitelist enough?

Reply:
Whitelist is enough, blacklisting * will stop everyone talking to the bot, including those on the whitelist.